config.txt
############################
### Ongoing Translations ###
############################

# Rebirth: Degenerate Slave Abuses Tyrant
python scraper.py https://www.scribblehub.com/read/266777-rebirth-degenerate-save-abuses-tyrant/chapter/266779/ "Rebirth Degenerate Slave Abuses Tyrant_重生之孽奴虐暴君_1-137_acidofanyi_138-210_ChrysanthemumBud_211-ongoing_Uhei_Cyrrene.txt" "//div[contains(@class, 'chp_raw')]" -head "//div[contains(@class, 'chapter-title')]" -b "//a[contains(@class, 'btn-next')]"
python scraper.py https://chrysanthemumbud.com/2020/06/19/rdsat-chapter-138-older-brother-will-protect-you-for-a-lifetime/ "Rebirth Degenerate Slave Abuses Tyrant_重生之孽奴虐暴君_1-137_acidofanyi_138-210_ChrysanthemumBud_211-ongoing_Uhei_Cyrrene.txt" "//div[contains(@class, 'entry-content')]" -head "//h1[contains(@class, 'entry-title')]" -rm "//p[a[contains(text(),'Next')]]" -f "DSAT" -b "(//a[contains(text(),'Next')])|(//a[@rel='next'])"
python scraper.py https://chrysanthemumbud.com/2020/06/23/rdsat-chapter-143-your-humble-servant-of-a-son-without-fortune/ "Rebirth Degenerate Slave Abuses Tyrant_重生之孽奴虐暴君_1-137_acidofanyi_138-210_ChrysanthemumBud_211-ongoing_Uhei_Cyrrene.txt" "//div[contains(@class, 'entry-content')]" -head "//h1[contains(@class, 'entry-title')]" -rm "//p[a[contains(text(),'Next')]]" -f "DSAT" -b "(//a[contains(text(),'Next')])|(//a[@rel='next'])"
python scraper.py https://uheitranslations.wordpress.com/2022/01/14/rdsat-chapter-211-lost-beads/ "Rebirth Degenerate Slave Abuses Tyrant_重生之孽奴虐暴君_1-137_acidofanyi_138-210_ChrysanthemumBud_211-ongoing_Uhei_Cyrrene.txt" "//div[@class='entry-content']" -head "//h1[contains(@class, 'entry-title')]" -f "DSAT" -b "//a[contains (@rel, 'next')]"

# Ghosts Know What I Experienced
python scraper.py https://spicychickentranslations.wordpress.com/novel-translations/gkwie/gk-chapter-1/ "Ghosts Know What I Experienced_1-65_total_72+8_SpicyChickenTranslations.txt" "//div[contains(@class, 'entry-content')]" -b "//a[contains(text(), '>>>')]"
python scraper.py https://spicychickentranslations.wordpress.com/novel-translations/gkwie/gk-chapter-61-part-2-real/ "Ghosts Know What I Experienced_1-65_total_72+8_SpicyChickenTranslations.txt" "//div[contains(@class, 'entry-content')]" -b "//a[contains(text(), '>>>')]"
python scraper.py https://spicychickentranslations.wordpress.com/novel-translations/gkwie/gkchapter-65-part-3_real/ "Ghosts Know What I Experienced_1-65_total_72+8_SpicyChickenTranslations.txt" "//div[contains(@class, 'entry-content')]" -b "//a[contains(text(), '>>>')]"

# The General’s Manor Young Concubine Survival Report
python scraper.py https://spicychickentranslations.wordpress.com/novel-translations/tgmycsr/tgm-chapter-1/ "The General’s Manor Young Concubine Survival Report_将军府小妾生存报告_SpicyChickenTranslations.txt" "//div[contains(@class, 'entry-content')]" -b "//a[contains(text(), '>>>')]"

# Superstar Aspirations
python scraper.py https://theflowergarden1412.wordpress.com/superstar-aspirations-sa-chapter-1/ "Superstar Aspirations_巨星问鼎[重生]_1-49_FlowerProvince_50-95_SurrealSlyTranslations_total165+12.txt" "//div[contains(@class, 'entry-content')]" -head "//h1[contains(@class, 'entry-title')]" -b "//a[contains(text(), 'Next')]"
python scraper.py https://theflowergarden1412.wordpress.com/superstar-aspirations-sa-chap-7/ "Superstar Aspirations_巨星问鼎[重生]_1-49_FlowerProvince_50-95_SurrealSlyTranslations_total165+12.txt" "//div[contains(@class, 'entry-content')]" -head "//h1[contains(@class, 'entry-title')]" -b "//a[contains(text(), 'Next')]"
python scraper.py https://surrealskytranslations.com/2020/03/21/sa-chapter-50-part-1/ "Superstar Aspirations_巨星问鼎[重生]_1-49_FlowerProvince_50-95_SurrealSlyTranslations_total165+12.txt" "//div[contains(@class, 'entry-content')]" -head "//h1[contains(@class, 'entry-title')]" -b "//a[contains(text(), 'Next')]"

# Panguan
python scraper.py https://tinysalttranslations.com/novels/ongoing/pg/pg-1/ "Panguan_判官_1-61_[ongoing_117+6]_TinySaltTranslations.txt"  "//div[@class = 'entry-content']" -head "//h1[@class = 'entry-title']" -rm "//p[@style='background-color:#ffeefa;']" -b "//a[contains(text(), 'Next')]"
python scraper.py https://tinysalttranslations.com/novels/ongoing/pg/pg-63/ "Panguan_判官_1-61_[ongoing_117+6]_TinySaltTranslations.txt"  "//div[@class = 'entry-content']" -head "//h1[@class = 'entry-title']" -rm "//p[@style='background-color:#ffeefa;']" -b "//a[contains(text(), 'Next')]"

# Copper Coins 铜钱龛世
# use all_p = soup.find_all(['p', 'h1', 'h2'])
python scraper.py https://tongqiankanshi.blogspot.com/2021/08/revised-tl-1-tong-qian-kan-shi-copper.html "Copper Coins_铜钱龛世_huxiyi.txt" "//div[contains(@class, 'entry-content')]"

# How is this Gentleman Feeling? / Is the Gentleman Feeling Alright?
python scraper.py https://uheitranslations.wordpress.com/2021/06/04/is-the-gentleman-feeling-alright-chapter-18/ "How is this Gentleman Feeling_君有疾否_ch18-36_[ongoing_total_89+3]_uhei.txt" "//div[@class='entry-content']" -head "//h1[contains(@class, 'entry-title')]" -f "Is the Gentleman Feeling Alright" -b "//a[span[contains (text(), 'Next')]]"

# Tiandi Baiju  (Fei Tian Ye Xiang)
python scraper.py https://beansprouttranslates.wordpress.com/2020/07/03/chapter-1-where-have-you-been-all-these-years/ "Tiandi Baiju_天地白驹_ch1-45[ongoing_total_74]_beansprout_grapeseed.txt" "//div[contains(@class, 'entry-content')][p]" -b "//a[contains(text(), 'Next')]" -head "//div[contains(@class, 'title-block')]//h1"

# Case File Compendium (Cookiecompany)
python scraper.py https://www.binganben.com/chapter-1-opening-shot/ "Case File Compendium_病案本_ch1-97_ONGOING_cookiecompany.txt" "//div[@class='entry-content']" -b "//a[span[contains (text(), 'Next')]]" -head "//h1[contains(@class, 'entry-title')]"
python bab_scrape.py https://www.binganben.com/chapter-97-no-longer-the-same/ "Case File Compendium_病案本_ch1-97_unlocked_ONGOING_cookiecompany.txt" "//div[@class='entry-content']" -b "//a[span[contains (text(), 'Next')]]" -head "//h1[contains(@class, 'entry-title')]"
# Case File Compendium (Tea Novel)
python scraper.py https://www.teanovel.com/case-file-compendium/chapter-1-the-mirror-opens/ "Case File Compendium_病案本_ch1-ONGOING_TeaNovel.txt" "//div[@class='entry-content']" -head "//h1[contains(@class, 'entry-title')]" -b "//div[contains(@class, 'nav-next')]/a"
python scraper.py https://teanovel.com/case-file-compendium/chapter-136-i-want-to-go-with-you/ "Case File Compendium_病案本_ch1-155_ONGOING_TeaNovel.txt" "//div[@class='entry-content']" -head "//h1[contains(@class, 'entry-title')]" -b "//div[contains(@class, 'nav-next')]/a"
# Case File Compendium (Meatbun) (Raw)
python scraper.py https://www.babayu.com/kanshu/89308_1.html "Case File Compendium_病案本_ch1-100_ONGOING_raw.txt" "//div[@id='BookText']" -head "//h1[contains(@class, 'article-title')]" -b "//a[contains(text(), '下')]"
python scraper.py https://www.babayu.com/kanshu/89308_100.html "Case File Compendium_病案本_ch1-100_ONGOING_raw.txt" "//div[@id='BookText']" -head "//h1[contains(@class, 'article-title')]" -b "//a[contains(text(), '下')]"
# Case File Compendium (ES)
python scraper.py https://www.wattpad.com/1114497307-libro-de-casos-cl%C3%ADnicos-cap%C3%ADtulo-1-espejo-roto "Case File Compendium_病案本_ES_Shiori_Atzuko.txt" "//div[@id='sticky-end']" -head  "//h1[@class='h2']" -b "//a[contains(@class, 'next-part-link')]"
python scraper.py https://www.wattpad.com/1196915439-libro-de-casos-cl%C3%ADnicos-parte-dos-cap%C3%ADtulo-198-sin "Case File Compendium_病案本_ES_Shiori_Atzuko.txt" "//div[@id='sticky-end']" -head  "//h1[@class='h2']" -b "//a[contains(@class, 'next-part-link')]"

# Right On Target (T97) (Raw)
python scraper.py https://www.babayu.com/kanshu/33931_1.html "Right On Target_准点狙击_ch1-110_ONGOING_raw.txt" "//div[@id='BookText']" -head "//h1[contains(@class, 'article-title')]" -b "//a[contains(text(), '下一')]"

#Saye
python scraper.py http://fantasy.888read.com/book/read/25035/1 Saye_撒野_mtl.txt "//div[@class='chapter_text']" -head "//h3[@class='chapter_title']" -b "//span[contains(@class,'next_chapter')]"
python scraper.py https://dogeballs.com/saye/chapter-111/ Saye_撒野_ch1-8_30-110_[ongoing_total_141+5]_AmidstWoods.txt "//div[contains(@class, 'post-content')]" -head //h1 -b "//a[@rel='next']"
python scraper.py https://haikushilan.wordpress.com/2019/01/21/saye-chapter-1/ Saye_撒野_ch1-63_[ongoing_total_141+5]_Sae.txt "//div[contains(@class, 'entry-content')]" -head "//h1[@class='entry-title']" -b "//a[contains(text(), 'NEXT')]"
python scraper.py https://haikushilan.wordpress.com/2022/03/16/saye-chapter-63/ Saye_撒野_ch1-63_[ongoing_total_141+5]_Sae.txt "//div[contains(@class, 'entry-content')]" -head "//h1[@class='entry-title']" -b "//a[contains(text(), 'NEXT')]"

# The Emperor's Strategy
# Everywhere in Juanhu is Beautiful
python scraper.py https://yorozuyaparty.com/dwgl-chapter-1/ "The Emperor's Strategy_帝王攻略_ch1-77_[ongoing_total_193+8]_yorozuya.txt"  "//div[contains(@class, 'entry-content')]" -head "//h1" -b "//a[@rel='next']"
python scraper.py https://yorozuyaparty.com/dwgl-chapter-44/ "The Emperor's Strategy_帝王攻略_ch1-77_[ongoing_total_193+8]_yorozuya.txt"  "//div[contains(@class, 'entry-content')]" -head "//h1" -b "//a[@rel='next']"
#use all_p = soup.find_all(['div']) instead of all_p = soup.find_all(['p'])
python scraper.py https://pucari.dreamwidth.org/651.html "Everywhere in Jianghu is Wonderful_帝王攻略_ch1-43_pucari.txt"  "//div[contains(@class, 'entry-content')]" -head "//h3" -b "//li[contains(@class, 'link_next')]/a"
python scraper.py https://yorozuyaparty.com/eijiw-chapter-34/ "Everywhere in Jianghu is Wonderful_江湖遍地是奇葩_ch5-51_[ongoing_total_172]_yorozuya.txt"  "//div[contains(@class, 'entry-content')]" -head "//h1" -b "//a[@rel='next']"

# All Under Heaven last ch 67
python scraper.py https://bookswithqianya.com/auh-ch1/ "All Under Heaven_天下_ongoing_qianya.txt" "//div[contains(@class, 'entry-content')]" -t no_tags -head  "//h1[contains(@class, 'entry-title')]" -b "//a[contains(text(), 'Next')]" -t no_tags
python scraper.py https://bookswithqianya.com/auh-chapter-67/ "All Under Heaven_天下_ongoing_qianya.txt" "//div[contains(@class, 'entry-content')]" -t no_tags -head  "//h1[contains(@class, 'entry-title')]" -b "//a[contains(text(), 'Next')]" -t no_tags

# The Plough
python scraper.py https://bookswithqianya.com/tp-ch1/ "The Plough_北斗_ongoing_qianya.txt" "//div[contains(@class, 'entry-content')]" -head  "//h1[contains(@class, 'entry-title')]" -t no_tags -b "//a[contains(text(), 'Next')]"
python scraper.py https://bookswithqianya.com/tp-67/ "The Plough_北斗_ongoing_qianya.txt" "//div[contains(@class, 'entry-content')]" -head  "//h1[contains(@class, 'entry-title')]" -t no_tags -b "//a[contains(text(), 'Next')]"

# 小白杨 My Little Poplar
python scraper.py https://blackbox-tl.com/novels/mlp/mlp-chapter-1/ "My Little Poplar_小白杨_ch1-16[ongoing_1-115+4]_AlexPT(blackbox-tl).txt" "//div[@class='entry-content']" -head "//h1[ contains(@class, 'entry-title')]" -rm "(//div[contains(@class, 'abh_box')])|(//p[a[contains(text(), 'Table of Contents')]])" -b "//a[contains(text(), 'Next Chapter')]"
python scraper.py https://nottegram.blogspot.com/2021/01/my-little-poplar-chapter-1-dilemma-of.html "My Little Poplar_小白杨_1-115+8_Nottegram.txt" "//div[contains(@class,'entry-content')]" -b "//a[contains(@class, 'blog-pager-newer-link')]"

# AWM: PUBG
python scraper.py http://blackbox-tl.com/novels/awm/awm-chapter-1/ "AWM PUBG_ch1-78[ongoing_86+4]_akarin_Coffee_Cherry_Lin_Wei(blackbox-tl).txt" "//div[@class='entry-content']" -head "//h1[ contains(@class, 'entry-title')]" -rm "(//div[contains(@class, 'abh_box')])|(//p[a[contains(text(), 'Table of Contents')]])" -b "//a[contains(text(), 'Next Chapter')]"
python scraper.py https://blackbox-tl.com/novels/awm/awm-chapter-41/ "AWM PUBG_ch1-78[ongoing_86+4]_akarin_Coffee_Cherry_Lin_Wei(blackbox-tl).txt" "//div[@class='entry-content']" -head "//h1[ contains(@class, 'entry-title')]" -rm "(//div[contains(@class, 'abh_box')])|(//p[a[contains(text(), 'Table of Contents')]])" -b "//a[contains(text(), 'Next Chapter')]"
python scraper.py https://blackbox-tl.com/novels/awm/awm-chapter-79/ "AWM PUBG_ch1-78[ongoing_86+4]_akarin_Coffee_Cherry_Lin_Wei(blackbox-tl).txt" "//div[@class='entry-content']" -head "//h1[ contains(@class, 'entry-title')]" -rm "(//div[contains(@class, 'abh_box')])|(//p[a[contains(text(), 'Table of Contents')]])" -b "//a[contains(text(), 'Next Chapter')]"

# Estranged
python scraper.py https://jinyutranslations.wordpress.com/2022/01/10/shsh92/ "Estranged_参商_ongoing_Jinyu.txt" "//div[contains(@class,'entry-content')]" --head "//h1[contains(@class, 'entry-title')]" -fn "//p[contains(@class,'has-small-font-size')]" -b "//a[contains(text(), 'next')]"
python scraper.py https://jinyutranslations.wordpress.com/2022/02/24/shsh105/ "Estranged_参商_ongoing_Jinyu.txt" "//div[contains(@class,'entry-content')]" --head "//h1[contains(@class, 'entry-title')]" -fn "//p[contains(@class,'has-small-font-size')]" -b "//a[contains(text(), 'next')]"

# First-Class Lawyer
python scraper.py https://sites.google.com/view/shandiantranslations/projects/the-lawyer/yjls-chapter-1 "First Class Lawyer_一级律师_1-6_shandian_6-ongoing_luckykoi(Chrysanthemum Garden).txt" "(//div[contains(@role, 'main')])" --b "//a[div/p[contains(text(), 'Next')]]
python scraper.py https://ninetysevenkoi.wordpress.com/2021/04/09/law-6p2/ "First Class Lawyer_一级律师_1-6_shandian_6-ongoing_luckykoi(Chrysanthemum Garden).txt" "//div[contains(@class,'entry-content')]" --head "//h1[contains(@class, 'entry-title')]"
python scraper.py https://chrysanthemumgarden.com/novel-tl/law/law-7/ "First Class Lawyer_一级律师_1-6_shandian_6-ongoing_luckykoi(Chrysanthemum Garden).txt" "//div[@class = 'entry-content']" -b "//a[span[text() = 'Next']]" -head "//span[@class = 'chapter-title'][1]"
python scraper.py https://chrysanthemumgarden.com/novel-tl/law/law-155/ "First Class Lawyer_一级律师_1-6_shandian_6-ongoing_luckykoi(Chrysanthemum Garden).txt" "//div[@class = 'entry-content']" -head "//span[@class = 'chapter-title'][1]" -b "//a[span[text() = 'Next']]"

# Thriller Trainee
python scraper.py https://chrysanthemumgarden.com/novel-tl/tltn/tltn-85/ "Thriller Trainee_惊悚练习生_1-ongoing_luckykoi(Chrysanthemum Garden).txt" "//div[@class = 'entry-content']" -head "//span[@class = 'chapter-title'][1]" -b "//a[span[text() = 'Next']]"

# Forces of Temptation
python scraper.py https://chrysanthemumgarden.com/novel-tl/fot/fot-1/ "Forces of Temptation_万有引力_1-87_ongoing_rhapsodite(Chrysanthemum Garden).txt" "//div[@class = 'entry-content']" -b "//a[span[text() = 'Next']]" -head "//span[@class = 'chapter-title'][1]"

# Lantern: Reflection of the Peach Blossoms
python scraper.py https://www.wattpad.com/739391698-lantern-reflection-of-the-peach-blossoms-chapter-1 "Lantern Reflection of the Peach Blossoms_提灯映桃花_1-11_edahkayno_12-49_ongoing_yunyun.txt" "//div[@id='sticky-end']" -b "//a[@aria-label='Continue reading next part']" -head  "//h1[@class='h2']"

config.py
############
### Raws ###
############

# To Rule in a Turbulent World (Raw)
output_filename = "亂世為王_raw_plaintext.txt"
first_ch_url = "http://www.jjwxc.net/onebook.php?novelid=3021665&chapterid=1"
next_ch_button_xpath = '//a[span[ text() = "下一章→"]]'
output_type = "plaintext"
body_xpath = '//*[@id="oneboolt"]/tbody/tr[2]/td[1]/div[1]'
chapter_header_xpath = "//h2"

# How is this gentleman faring (Raw)
output_filename = "君有疾否_raw_plaintext.txt"
first_ch_url = "http://www.jjwxc.net/onebook.php?novelid=2862999&chapterid=1"
next_ch_button_xpath = '//a[span[ text() = "下一章→"]]'
output_type = "plaintext"
body_xpath = '//*[@id="oneboolt"]/tbody/tr[2]/td[1]/div[1]'
chapter_header_xpath = "//h2"

# Twin Jades of Jiangdong (Raw)
output_filename = "江东双璧_raw_plaintext.txt"
first_ch_url = "http://www.jjwxc.net/onebook.php?novelid=2467979&chapterid=1"
next_ch_button_xpath = '//a/span[ text() = "下一章→"]'
output_type = "plaintext"
body_xpath = '//*[@id="oneboolt"]/tbody/tr[2]/td[1]/div[1]'
chapter_header_xpath = "//h2"

# Royal Road (Raw)
output_filename = "簪缨问鼎_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/43856.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# Transmigrating to the Ming Dynasty’s Imperial Examination  (Raw)
output_filename = "穿到明朝考科举_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/100997.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# Qiang Jin Jiu (Raw)
output_filename = "将进酒_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/65246.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# 2ha (Raw)
output_filename = "二哈和他的白猫师尊_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/33386.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# Yuwu (Raw)
output_filename = "余污_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/66722.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# TGCF (Raw)
output_filename = "天官赐福_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/6240.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# MDZS (Raw)
output_filename = "魔道祖师_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/5419.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# Breaking Through the Clouds (Raw)
output_filename = "破云_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/6247.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# Scum Villain (Raw)
output_filename = "人渣反派自救系统_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/7118.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# A Sword of Frost (raw)
output_filename = "一剑霜寒_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/82144.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# Beidou (raw)
output_filename = "北斗_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/146845.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# Panguan (raw)
output_filename = "判官_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/167128.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# 九霄 (Meng Xi Shi) (Raw)
output_filename = "九霄_raw_plaintext.txt"
first_ch_url = "https://www.zhenhunxiaoshuo.com/75131.html"
next_ch_button_xpath = "//span[@class='article-nav-next']/a"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1"

# Yingnu FTYX (Raw, no extras?)
output_filename = "鹰奴_raw_plaintext.txt"
first_ch_url = "https://www.lwtxt.net/37015/890751.html"
next_ch_button_xpath = "//div/a[text()='下一章']"
output_type = "plaintext"
body_xpath = "//div[@id='content']"
chapter_header_xpath = "//h2"

# 天下 MXS (Raw)
output_filename = "天下_raw_plaintext.txt"
first_ch_url = "http://www.lwtxt.net/37358/1147658.html"
next_ch_button_xpath = "//div/a[text()='下一章']"
output_type = "plaintext"
body_xpath = "//div[@id='content']"
chapter_header_xpath = "//h2"

# 参商 MSX (Raw) #there's no chapter header, so comment out chapter dividing line and header
output_filename = "参商_raw_plaintext.txt"
first_ch_url = "https://www.52shuku.vip/jiakong/pm/h2HD_2.html"
next_ch_button_xpath = "//a[text()='下一页']"
output_type = "plaintext"
body_xpath = "//article[@class='article-content']"
chapter_header_xpath = "//h1[1]"

# Qiang Jin Jiu (Raw, Trad?)
output_filename = "将进酒_cn_trad_raw_plaintext.txt"
first_ch_url = "https://www.twfanti.com/JiangJinJiu/read_1.html"
next_ch_button_xpath = "//a[@class='pt-nextchapter']"
output_type = "plaintext"
body_xpath = "//div[@class='size16 color5 pt-read-text']"
chapter_header_xpath = "//h1/a[@title]"

#################
## Translated ###
#################

#faraway wanderers
first_ch_url = "https://sparklingwatertrans.wordpress.com/2018/11/23/faraway-wanderers-chapter-1"
first_ch_url = "https://sparklingwatertrans.wordpress.com/2019/04/08/faraway-wanderers-chapter-21/"
next_ch_button_text = "Next Chapter"
next_ch_button_xpath = '//a[text() = "Next Chapter"]'
output_type = "plaintext"
body_xpath = "//article"
chapter_header_xpath = "h1"
output_filename = "天涯客_ch1-30_plaintext.txt"


#qian qiu
first_ch_url = "https://snowycodex.com/translations/novels/ta/chapter-1/"
next_ch_button_text = "Next Chapter"
next_ch_button_xpath = '//a[text() = "Next Chapter"]'
output_type = "plaintext"
body_xpath = "//article/div"
chapter_header_xpath = "h2"
output_filename = "千秋_ch1-69_Snowy_Codex_plaintext.txt"

first_ch_url = "https://qianqiunoveltranslation.wordpress.com/2021/05/17/chapter-91/"
next_ch_button_xpath = '//a[@rel="next"]'
output_type = "plaintext"
body_xpath = "//article/div"
chapter_header_xpath = "h1"
output_filename = "千秋_ch91-119ongoing_Jul20_Superglue_plaintext.txt"

# those years in quest of honor mine
first_ch_url = "https://perpetualdaydreams.com/novel/tyqhm/tyqhm-1/"
next_ch_button_xpath = '//a[text() ="NEXT>"]'
output_type = "plaintext"
body_xpath = "//article/div"
chapter_header_xpath = "h1"
output_filename = "当年万里觅封侯_ch104+2extras_perpetual_daydreams_plaintext.txt"

#Wushuang
first_ch_url = "https://dustbunnytranslations.wordpress.com/2020/01/08/wushuang-chapter-79/"
next_ch_button_xpath = '(//article/div[@class="entry-content"]/p/a[last()])[last()]'
output_type = "plaintext"
body_xpath = "//article/div[@class='entry-content']"
chapter_header_xpath = "//h1[@class='entry-title']/a"
output_filename = "无双_dust_bunny_translations_plaintext.txt"


# Sha Po Lang #Note: Next button links are broken
first_ch_url = "https://www.northwestflower.net/sha-po-lang/prologue" #"https://www.northwestflower.net/sha-po-lang/extras"
next_ch_button_xpath = '(//a[@class="FKF6mc TpQm9d QmpIrf" or @class="FKF6mc TpQm9d xkUom"])[last()]'
output_type = "plaintext"
body_xpath = "(//div[@class='tyJCtd mGzaTb baZpAe'])[2]"
chapter_header_xpath = "(//div[@class='tyJCtd mGzaTb baZpAe'])[1]"
output_filename = "杀破狼_northwestflower_plaintext.txt"

#Liu Yao

# ch v1c1-v1c8, Nanming
output_type = "plaintext"
output_filename = "六爻_ch1-8_nanming_raw_plaintext.txt"
first_ch_url = "https://nanming.wordpress.com/2017/09/23/liuyaov1c1/"
next_ch_button_xpath = "//a[@rel='next']"
body_xpath = "//div[@class='entry-content']"
chapter_header_xpath = "//h1"

# ch v1c9, Blob Translations
output_type = "plaintext"
output_filename = "六爻_ch9-33_bobtranslations_raw_plaintext.txt"
first_ch_url = "https://www.blobtranslations.com/liu-yao-the-revitalization-of-fuyao-sect/volume-%E2%85%A0-chapter-9/"
next_ch_button_xpath = "//a[text()= ' [Next Chapter]']"
body_xpath = "//div[@class='entry-content']"
chapter_header_xpath = "//h1"

# ch 34-68, Pizzilations
output_type = "plaintext"
output_filename = "六爻_ch34-68_piizzilations_raw_plaintext.txt"
first_ch_url = "https://pizzilations.com/liu-yao-chapter-34/"
next_ch_button_xpath = "//a[text() = 'Next →']"
body_xpath = "//div[@class='entry-content']"
chapter_header_xpath = "//h1"

# ch 68-85, Rustycol
output_type = "plaintext"
output_filename = "六爻_ch34-68_rustycol_raw_plaintext.txt"
first_ch_url = "https://rustycol.dreamwidth.org/951.html"
next_ch_button_xpath = "//a[img [@alt='Next Entry']]"
body_xpath = "//div[@class='entry-content']"
chapter_header_xpath = "(//a[@title])[1]"
# ch 85-109 Chichi
output_type = "plaintext"
output_filename = "六爻_ch85-109_chichi_raw_plaintext.txt"
first_ch_url = "https://chichilations.home.blog/2021/03/22/ly-85/"
next_ch_button_xpath = "//a[ text()='Next']"
body_xpath = "//div[@class='entry-content']"
chapter_header_xpath = "//h1[@class='entry-title']"

#Can Ci Pin ch 2 - 48 Golden Plastic Sisters
output_type = "plaintext"
output_filename = "殘次品_prologue-ch48_golden_plastic_sisters_raw_plaintext.txt"
first_ch_url = "https://www.wattpad.com/613294385-can-ci-pin-imperfections-%E6%AE%8B%E6%AC%A1%E5%93%81-bl-by-priest-prelude"
next_ch_button_xpath = "//a[@aria-label='Continue reading next part']"
body_xpath = "//div[@id='sticky-end']"
chapter_header_xpath = "//h1[@class='h2']"

#Jin Se Wangxianbot ch 1-10
output_type = "plaintext"
output_filename = "锦瑟_ch1-10_wangxianbot_plaintext.txt"
first_ch_url = "https://www.wattpad.com/754966819-jin-se-priest-english-fan-translation-bl-novel"
next_ch_button_xpath = "//a[@aria-label='Continue reading next part']"
body_xpath = "//div[@id='sticky-end']"
chapter_header_xpath = "//h1[@class='h2']"

#Advance Bravely She Bu Ke Dang @anhnguyen12332
output_type = "plaintext"
output_filename = "势不可挡_ch1-224_@anhnguyen12332_plaintext.txt"
first_ch_url = "https://www.wattpad.com/299917904-advance-bravely-shi-bu-ke-dang-%E5%8A%BF%E4%B8%8D%E5%8F%AF%E6%8C%A1-1-kickboxing"
next_ch_button_xpath = "//a[@aria-label='Continue reading next part']"
body_xpath = "//div[@id='sticky-end']"
chapter_header_xpath = "//h1[@class='h2']"

# 这锅我不背 I'm Not Shouldering This Blame
first_ch_url = "https://rainbow-reads.com/stb-index/stb-chapter-1/"
next_ch_button_xpath = "//a[text()='Next Chapter']"
output_type = "plaintext"
body_xpath = "//div[@class='entry-content']"
chapter_header_xpath = "//h1[@class='entry-title']"
output_filename = "这锅我不背_ch1-12_Rainbow_Turtle_Translations_plaintext.txt"

first_ch_url = "https://pickupnovels.com/im-not-shouldering-this-blame/chapter-13-dumbfounded-circle/"
next_ch_button_xpath = "//div[@class='wp-post-navigation-next']/a"
output_type = "plaintext"
body_xpath = "//div[@class='entry-content']"
chapter_header_xpath = "//h1[@class='entry-title']"
output_filename = "这锅我不背_ch13-175_Chas_Spoilerish_Translations_plaintext.txt"

####################
### No Footnotes ###
####################

#Jin Se Crysanthemum Garden
output_type = "plaintext"
output_filename = "锦瑟_ch8-79[end]+2extras_halfdeadhulijing(Crysanthemum Garden)_plaintext.txt"
first_ch_url = "https://chrysanthemumgarden.com/novel-tl/js/js-8/"
next_ch_button_xpath = "//a[span[text() = 'Next']]"
body_xpath = "//div[@class = 'entry-content']"
chapter_header_xpath = "//span[@class = 'chapter-title'][1]"

#Qiang Jin Jiu_Lianyin
output_type = "plaintext"
output_filename = "将进酒_ch1-ongoing_Lianyin_plaintext.txt"
first_ch_url = "https://cangji.net/qiangjinjiu/qiangjinjiu-c146/"
next_ch_button_xpath = "(//div[@class='post-entry']/center/a[img])[last()]"
body_xpath = "//div[@class = 'post-entry']"
chapter_header_xpath = "//h1"

# Fei Pin Ying Qiang
first_ch_url = "https://perpetualdaydreams.com/novel/fpyq/fpyq-1"
next_ch_button_xpath = '//a[text() ="NEXT>"]'
output_type = "plaintext"
body_xpath = "//article/div[@class='post-content']"
chapter_header_xpath = "h1"
output_filename = "妃嫔媵嬙_ch1-76+1extra_perpetual_daydreams_NOTRANSLATORSNOTES_plaintext.txt"


